addons/web_editor/static/src/js/editor/odoo-editor/src/powerbox/Powerbox.js

```diff
@@ -282,14 +282,18 @@ export class Powerbox {
         if (this.getContextFromParentRect) {
             options['parentContextRect'] = this.getContextFromParentRect();
         }
-        const position = getRangePosition(this.el, this.document, options);
-        if (position) {
-            let { left, top } = position;
-            this.el.style.left = `${left}px`;
-            this.el.style.top = `${top}px`;
-        } else {
-            this.hide();
-        }
+        let initialWidth;
+        do {
+            initialWidth = this.el.offsetWidth;
+            const position = getRangePosition(this.el, this.document, options);
+            if (position) {
+                let { left, top } = position;
+                this.el.style.left = `${left}px`;
+                this.el.style.top = `${top}px`;
+            } else {
+                this.hide();
+            }
+        } while (this.el.offsetWidth !== initialWidth);
```

Comment:
When pressing arrowdown, we call `_render` here https://github.com/odoo/odoo/blob/27f431acaab985a7c06496490bd86dd698b478d2/addons/web_editor/static/src/js/editor/odoo-editor/src/powerbox/Powerbox.js#L394 , which calls `_resetPosition`. The range doesn't change, but after every calculation of the position , we expect to have `marginRight=20px` but it gets auto increased because of `max-width=100%` so at each render we will increase the width by `20px` until it's finally in position correctly, that's why I added the loop, It's a bit hacky solution

Reply:
I'm still quite confused because it seems way too convoluted. Maybe @Goaman can shed some light on this?

Reply:
I've talked in private with @Mtaylorr and we found out that the fix could be done in css

Reply:
I udpated the fix , Thanks

Reply:
I knew I could count on you, thanks! 👍